### ISSUE TITLE
if el.classList not contains directiveClass callback function will not process

### ISSUE THREAD

**Issue by chnkarl** [alexshink/vue3-clickaway#1]
origin 
```
if ( elementIsActive ) {
    el.classList.add(directiveClass);
  } else  if ( el.classList.contains(directiveClass) ) {
    el.classList.remove(directiveClass);
    callback({el, event});
  }
```

work for me
```
if ( elementIsActive ) {
    el.classList.add(directiveClass);
  } else {
    if ( el.classList.contains(directiveClass) ) {
      el.classList.remove(directiveClass);
    }
    callback({el, event});
  }
```

**Comment by alexshink** [comment]
Hello @chnkarl!

This plugin is designed to work that way. The callback should only fire if the element was active. Otherwise, the callback will always fire (even in the case when we did not interact with the element.).

Can you share in which case you need this behavior of the plugin? I will consider this case if it can really be useful - I can add an additional option that will enable this behavior.

**Comment by chnkarl** [comment]
thanks for your reply！

I have a vue3 component，it's a color picker for vue3(which name is slim-color-picker), someone can use this component to change color.&nbsp;


when someone click a button, the color-picker will show the modal, but he/she may not click modal, then just hide the modal,&nbsp; at that time, your logic is not work for me, because if someone want to hide modal, he/she must click&nbsp; the modal before hiding.


but i really appreciate your component, thanks very much.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Alex ***@***.***&gt;; 
发送时间: 2022年3月9日(星期三) 凌晨3:11
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alexshink/vue3-clickout] if el.classList not contains directiveClass callback function will not process (Issue #1)





 
Hello @chnkarl!
 
This plugin is designed to work that way. The callback should only fire if the element was active. Otherwise, the callback will always fire.
 
Can you share in which case you need this behavior of the plugin? I will consider this case if it can really be useful - I can add an additional option that will enable this behavior.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by alexshink** [comment]
Hey @chnkarl!

Thanks for the detailed explanation and I'm glad the directive helped you somehow. This is an interesting case. This really makes sense. Of course, I would recommend implementing it in a different way, because the directive was originally designed to solve a slightly different task.

**For fun:** if you do not change anything in the directive, then such a stupid way came to my mind: add a ref to your component and after you open your picker, you can artificially click on the target element on which the directive enabled, this will cause the directive to fire and everything will work as it should: ` this.$refs.yourRefName.$el.click()` 

**Now seriously:** this is a great opportunity to extend the functionality of the directive, which I already enjoy doing, thanks to you) Within a few hours I will publish an update of the package with new features, including your case and adding all the details to the documentation.

**Comment by alexshink** [comment]
Hello again, @chnkarl!

I just released a new version 1.1.0 with backwards compatibility and updated documentation, please visit this section for a review: 
https://github.com/alexshink/vue3-clickout#arguments 

If you still have a need - try to use this. I added functionality for your needs and made one more additional feature. For now, I leave this issue open, I will be glad for feedback.
